types/simulation/types.go

```diff
@@ -39,7 +41,7 @@ type WeightedProposalMsg interface {
 	MsgSimulatorFn() MsgSimulatorFn // msg simulator function
 }
 
-type MsgSimulatorFn func(r *rand.Rand, ctx sdk.Context, accs []Account) sdk.Msg
+type MsgSimulatorFn func(r *rand.Rand, ctx sdk.Context, accs []Account, cdc address.Codec) (sdk.Msg, error)
```

Comment:
Breaking this is quite disruptive for all modules implementing simulation. Is there a way around it? After a quick skim I couldn't find for which simulation it is actually used.

If not, we definitely need a section under Modules > All > Simulation in the UPGRADING.md

Reply:
I couldn't find a way to avoid it other than using a global codec for simulation. In the end, either the prefix or the codec must be passed all the way through the stack.

Reply:
but for which simulation is that param used? I cannot find it 😅

Reply:
right now just for the staking https://github.com/cosmos/cosmos-sdk/pull/19742/files#diff-ccc155cefb86edd2f89c6436d48cde5df2e5734fa9e7df73fd543a29d318b92dR36-R52

But It'll be updated in coming PRs.